codes/zig/include/TreeNode.zig

```diff
@@ -28,32 +28,26 @@ pub fn arrToTree(comptime T: type, mem_allocator: std.mem.Allocator, list: []T)
     if (list.len == 0) return null;
     var root = try mem_allocator.create(TreeNode(T));
     root.init(list[0]);
-
-    const TailQueue = std.TailQueue(?*TreeNode(T));
-    const TailQueueNode = std.TailQueue(?*TreeNode(T)).Node;
-    var que = TailQueue{};
-    var node_root = TailQueueNode{ .data = root };
-    que.append(&node_root); 
+    var que = std.ArrayList(*TreeNode(T)).init(std.heap.page_allocator);
```

Comment:
Is this a List or Heap?

Reply:
> Is this a List or Heap?

The codes were updated. 
Now， it supports TailQueue, and also support ArrayList to work as queue.